examples/microwave-oven-app/microwave-oven-common/src/microwave-oven-device.cpp

```diff
@@ -35,38 +35,79 @@ void ExampleMicrowaveOvenDevice::MicrowaveOvenInit()
     mOperationalStateInstance.Init();
     mMicrowaveOvenModeInstance.Init();
     mMicrowaveOvenControlInstance.Init();
+
+    // set default value for attribute SelectedWattIndex and WattRating
+    if (mMicrowaveOvenControlInstance.HasFeature(MicrowaveOvenControl::Feature::kPowerInWatts))
+    {
+        static_assert(ArraySize(mWattSettingList) > 0, "Watt setting list is empty!");
+        mSelectedWattIndex = ArraySize(mWattSettingList) - 1;
+        mWattRating        = mWattSettingList[mSelectedWattIndex];
+    }
+    else
+    {
+        mWattRating = kExampleWatt5;
+    }
 }
 
 /**
  * MicrowaveOvenControl cluster
  */
 Protocols::InteractionModel::Status
-ExampleMicrowaveOvenDevice::HandleSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTime, uint8_t powerSetting)
+ExampleMicrowaveOvenDevice::HandleSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTimeSec, bool startAfterSetting,
+                                                               Optional<uint8_t> powerSettingNum,
+                                                               Optional<uint8_t> wattSettingIndex)
 {
     // placeholder implementation
     Status status;
+    // update cook mode
     if ((status = mMicrowaveOvenModeInstance.UpdateCurrentMode(cookMode)) != Status::Success)
     {
         return status;
     }
-    mMicrowaveOvenControlInstance.SetCookTime(cookTime);
-    mMicrowaveOvenControlInstance.SetPowerSetting(powerSetting);
+
+    mMicrowaveOvenControlInstance.SetCookTimeSec(cookTimeSec);
+
+    // if using power as number, check if powerSettingNum has value before setting the power number
+    if (powerSettingNum.HasValue())
+    {
+        mPowerSettingNum = powerSettingNum.Value();
```

Comment:
spec also says `[8.140] If this field is missing, the PowerSetting attribute SHALL be set to 100 if MaxPower is not supported by the server, otherwise it SHALL be set to MaxPower if the MaxPower attribute is supported by the server.`

Is this done anywhere here? I would have expected the logic related to this to be near here ... otherwise the comment to explain.

Reply:
the logic is implemented  in the SDK,  if the field is missing, the given powerSettingNum here is the max power value, will   add some comments on this.